gapic-generator/lib/gapic/path_pattern/parser.rb

```diff
@@ -15,66 +15,131 @@
 # limitations under the License.
 
 require "gapic/path_pattern/segment"
+require "gapic/path_pattern/pattern"
 
 module Gapic
   module PathPattern
+    ##
     # A path pattern parser.
-    # see https://google.aip.dev/122, https://google.aip.dev/123
+    # takes a pattern and transforms it into a collection of parsed segments
+    # @see https://google.aip.dev/122
+    # @see https://google.aip.dev/123
     #
-    # @!attribute [r] path_pattern
-    #   @return [String] The path pattern to be parsed.
-    # @!attribute [r] segments
-    #   @return [Array<Segment|String>] The segments of the parsed path pattern.
-    class Parser
+    module Parser
+      ##
+      # @param path_pattern [String] The path pattern to be parsed
+      # @return [Gapic::PathPattern::Pattern]
+      def self.parse path_pattern
+        remainder = path_pattern.sub(%r{^\/}, "").sub(%r{\/$}, "")
```

Comment:
If you use `%r` for regexes, you don't need to escape forward slashes.

This also applies to a number of additional inline regexes further down.

Reply:
fixed for all %r{}, still have to escape for %r//